Create Request Model

nicholasrwx opened the pull request:
### What this PR does (required):
- This PR address the creation of the `Request Model` as per #9
- its the Model for database Queries Via the `request` routes
- This is for fetching all requests associated with the logged in user, creating new requests, and updating existing requests.

### Any information needed to test this feature (required):
- To test this feature you would have to run the  request `API Calls` in the `client/helpers/APICalls` folder, with test data, as they are not connect to any form or button in the app yet.

ezeilo-su reviewed (changes requested):
Please make those changes and commit them for another review.

ezeilo-su commented on server/models/Request.js:
```diff
@@ -0,0 +1,27 @@
+  sitter_id: {
+    type: String,
+  },
+  start: {
+    type: Number,
+  },
+  end: {
+    type: Number,
+  },
```
These ones should be of type `Date` with a default value. Please refer to the issue.

nicholasrwx replied:
These changes have now been made and approved in https://github.com/hatchways/team-igneous/pull/56/commits/8ade7c03ebe6f587d3dc02b8f8aa85f5e9dcf490 :)

ezeilo-su reviewed (changes requested):
Please implement the requested changes for another review.

ezeilo-su commented on server/models/Request.js:
```diff
@@ -0,0 +1,27 @@
+  end: {
+    type: Number,
+  },
+  accepted: {
+    type: Boolean,
+  },
+  declined: {
+    type: Boolean,
+  },
+  paid: {
+    type: Boolean,
+  },
```
These ones should have a default value. Please refer to the issue.

nicholasrwx replied:
These changes have now been made and approved in https://github.com/hatchways/team-igneous/pull/56/commits/8ade7c03ebe6f587d3dc02b8f8aa85f5e9dcf490 :)

ezeilo-su reviewed (changes requested):
Make those changes, please. :+1:

nicholasrwx commented:
I have addressed all the issues above, in this commit https://github.com/hatchways/team-igneous/pull/56/commits/615b2d25bf875cd32396ea2566226b9128d00f9d

nicholasrwx commented:
I just added a couple extra values to the model earlier I didn't have a moment to push until now.

nicholasrwx commented on server/models/Request.js:
```diff
@@ -0,0 +1,27 @@
+  },
+  sitter_id: {
+    type: String,
+  },
```
These changes have now been made and approved in https://github.com/hatchways/team-igneous/pull/56/commits/8ade7c03ebe6f587d3dc02b8f8aa85f5e9dcf490 :)
